### DISCUSSION TITLE
`GetDCEx` is missing from `win32gui`

### DISCUSSION THREAD

**Discussion by xa1on** [mhammond/pywin32#2718] [Q&A] [answered]
Feature Request: [`win32gui.GetDCEx`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getdcex)
I can't seem to find the `GetDCEx` function in the `win32gui` module. I'm creating a program that uses the device content of a window for an image sequence that is able to clip the visible area of the parent window and the only way to do this is with the `GetDCEx` function. `GetDCEx` has a lot of different options that are unavailable with `GetDC` alone so I was just wondering if there would be any plans to support it.
I've noticed a lot of references in the repo to `GetDCEx`, however, it's not in the pywin32 [documentation](https://mhammond.github.io/pywin32/win32gui.html).

**Comment by Avasam** [comment, marked as answer]
@xa1on If you can get a `PyCWnd`, then you can call `GetDCEx` on it.
For example:
```py
>>> from pythonwin.win32ui import GetForegroundWindow
>>> wnd = GetForegroundWindow()
>>> wnd
object 'PyCWnd' - assoc is 0000027D6FCFDAB0, vi=<None>, notify=0,ch/u=0/0, mh=0, kh=0
>>> wnd.GetDCEx(None, 0)
object 'PyCDC' - assoc is 0000027D706CB968, vi=<None>
```

typeshed can help you find a couple ways of getting a `PyCWnd` (type stubs as documentation! :D ) https://github.com/search?q="->+PyCWnd"+OR+"->+_win32typing.PyCWnd"+repo%3Apython%2Ftypeshed&type=code
